android/src/main/kotlin/com/lahaus/iterable_flutter/NotificationParser.kt

```diff
@@ -0,0 +1,46 @@
+package com.lahaus.iterable_flutter
+
+import java.util.HashMap
+
+class NotificationParser {
+
+  fun parse(dataPush: Map<String, Any?>): Map<String, Any?> {
+
+    val mapPushData = getAllFormatInt(dataPush)
+    return buildSendPushMap(mapPushData)
+  }
+
+  private fun getAllFormatInt(dataPush: Map<String, Any?>): Map<String, Any?>{
+    val map: MutableMap<String, Any?> = HashMap()
+    for ((key, value) in dataPush) {
+
+      if(value is String){
+
+        if(value.intOrString() is Int){
+          val newValueInt: Int = value.toInt();
+          map[key] = newValueInt;
+        }else{
+          map[key] = value;
+        }
+      }
+
+    }
+    return  map;
+  }
+
+  private fun buildSendPushMap(mapPushData: Map<String, Any?>): Map<String, Any?> {
+    return mapOf(
+      "title" to mapPushData["title"],
+      "body" to mapPushData["body"],
+      "additionalData" to mapPushData
+    )
+  }
+
+}
+
+
+fun String.intOrString() = try { // returns Any
+  toInt()
+} catch(e: NumberFormatException) {
+  this
+}
```

Comment:
I'm not sure if this should be the best approach, probably something like (I'm not sure if it compiles well, but I feel like (consider) putting logic into a `catch` is not a good practice)

```kotlin
when(myString) {
    is String -> myString.toInt()
    else -> this
}
```

Or doing the logic taking into consideration this stdlib extension:
[https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/to-int-or-null.html](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/to-int-or-null.html)